spec/requests/host_aggregates_spec.rb

```diff
@@ -67,6 +68,61 @@
     end
   end
 
+  describe 'POST /api/host_aggregates/:id' do
+
+    before do
+      @zone = FactoryBot.create(:zone, :name => "api_zone")
+      @microstack = FactoryBot.create(:ems_cloud)
+      @provider = FactoryBot.create(:provider, :zone => @zone)
+      @microstack.update(:provider => @provider) ## microstack.provider = provider 
+      @infra = FactoryBot.create(:ems_infra, :name => "microstack ironic", :hostname => "ironic", :zone => @zone, :provider => @provider)
+      @cluster = FactoryBot.create(:ems_cluster, :name => "Microstack Compute", :ext_management_system => @infra)
+      @host = FactoryBot.create(:host_with_authentication, :ext_management_system => @infra, :ems_cluster => @cluster, :name => "host", :hostname => "host.ironic.microstack", :vmm_vendor => "openstack_infra")
+    end
+
+    it 'adds a host to the host aggregate' do
+      host_aggregate = FactoryBot.create(:host_aggregate)
```

Comment:
You can set the ext_management_system here instead of `@microstack.update(:host_aggregates => [host_aggregate])` below:
```suggestion
      host_aggregate = FactoryBot.create(:host_aggregate, :ext_management_system => ems)
```